content/en/blog/docker-multi-arch/index.md

```diff
@@ -0,0 +1,114 @@
+---
+date: 2021-09-20
+title: "The Arm Adventure on Docker"
+linkTitle: "The Arm Adventure on Docker"
+description: "Since the Findy Agency project launched, Docker has been one of our main tools to help set up the agency development and deployment environments. Unexpected headache developed when our colleague purchased a M1 Mac and our images refused to run on ARM platform."
+author: Laura Vuorenoja
+resources:
+- src: "**.{png,jpg}"
+  title: "Image #:counter"
+---
+
+Since the Findy Agency project launched, Docker has been one of our main tools to help set up the agency development and deployment environments. First of all, we use Docker images for our cloud deployment. On a new release, the CI build pipeline bundles all needed binaries to each service image. After the build, the pipeline pushes Docker images to [the GitHub container registry](https://github.blog/2020-09-01-introducing-github-container-registry/), from where the deployment pipeline fetches them and updates the cloud environment. 
+
+{{< imgproc agency-deployment-pipeline Fit "925x925" >}}
+<em>Agency deployment pipeline: New release triggers image build in GitHub actions. When the new image is in the registry, AWS Code Pipelines handles the deployment environment update.</em>
+{{< /imgproc >}}
+
+
+In addition, we've used Docker to take care of the service orchestration in a local development environment. Agency consists of three different services, and if one wishes to set agency fully up on a local desktop, it is straightforward by executing a docker-compose script. The script pulls correct images to the local desktop and sets needed configuration parameters. Without the orchestration, setting up and updating the three services would be cumbersome.
+
+{{< imgproc arch-overview Fit "625x625" >}}
+<em>High-level architecture of Findy Agency. Setting up agency to localhost is most straightforward with the help of container orchestration tool.</em>
+{{< /imgproc >}}
+
+Until recently, we were happy with our image-building pipeline. Local environments booted up with ease, and the deployment pipeline rolled out updates beautifully. Then one day, our colleague with an M1-equipped Mac tried out the docker-compose script. Running the agency images in an Arm-based architecture was something we hadn't considered. We built our Docker images for amd64 architecture, while M1 Macs expect container images for arm64 CPU architecture. It became clear we needed to support also the arm64, as we knew that the popularity of the M1 chipped computers would only increase in the future.
+
```

Comment:
I would be very interested to know what is the current situation with Docker and Go on M1. I would be even interested to know some highly detailed technical points that we are losing now when we have multiple CPU architectures. All of this could be explained with excellent references, no need to write here.
Summary: In this chapter, I would no I am interested to know what happened for Docker and for Go? And how Docker was running Linux containers before.